src/core/auth/qgsauthconfigurationstorage.h

```diff
@@ -0,0 +1,462 @@
+/***************************************************************************
+  qgsauthconfigurationstorage.h - QgsAuthConfigurationStorage
+
+ ---------------------
+ begin                : 20.6.2024
+ copyright            : (C) 2024 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSAUTHCONFIGURATIONSTORAGE_H
+#define QGSAUTHCONFIGURATIONSTORAGE_H
+
+
+#include "qgis_sip.h"
+#include "qgis_core.h"
+#include "qgis.h"
+#include "qgsauthconfig.h"
+#include "qgsauthcertutils.h"
+
+#include <QString>
+#include <QObject>
+
+
+/**
+ * \ingroup core
+ *  Abstract class that defines the interface for all authentication configuration storage implementations.
+ *  \since QGIS 3.40
+ */
+class CORE_EXPORT QgsAuthConfigurationStorage: public QObject SIP_ABSTRACT
+{
+    Q_OBJECT
```

Comment:
Do we need sip casting code here? Or do we want to hide the internal classes for safety?

Reply:
I think the if we need casting I did not design the API correctly. 
For custom storages used by plugins we probably don't need it anyway because the plugin will keep an handle to the its own managed implementation.